wayland-client/src/proxy.rs

```diff
@@ -300,6 +313,15 @@ where
     }
 }
 
+impl<I: Interface> Debug for Main<I>
+where
+    I: Debug + AsRef<Proxy<I>> + From<Proxy<I>>,
+{
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{:?}[MAIN]", self.inner)
```

Comment:
Given `Main` contains an `Attached`, we may want to directly defer to the underlying Proxy (`self.inner.inner`).